### PR TITLE
[tests-only][full-ci] added test to reshare resources to other users SharingNg

### PR DESCRIPTION
**Pull request by PrajwolAmatya**
## Description
In this PR, added test to reshare a resource to other users.

Added Scenarios:
```feature
Scenario Outline: reshare a file to user with different roles
Scenario Outline: reshare a folder to user with different roles
Scenario Outline: reshare a file inside project space to a user with different roles
Scenario Outline: reshare a folder inside project space to a user with different roles
```

## Related Issue

- https://github.com/owncloud/ocis/issues/7947

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiSharingNg/reshare.feature [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+      | shareType       | user                       |
+      | permissionsRole | <reshare-permissions-role> |
+    Then the HTTP status code should be "200"
+    And for user "Carol" the space Shares should contain these entries:
+      | textfile1.txt |
+    Examples:
+      | permissions-role | reshare-permissions-role |
+      | Viewer           | Viewer                   |
+      | File Editor      | Viewer                   |
+      | File Editor      | File Editor              |
```
Comment (SagarGi) [inline tests/acceptance/features/apiSharingNg/reshare.feature:39]:
```suggestion
      | permissions-role | reshare-permissions-role |
      | Viewer           | Viewer                   |
      | File Editor      | Viewer                   |
      | Viewer           | File Editor              |
      | File Editor      | File Editor              |
```
combination if needed?

File: tests/acceptance/features/apiSharingNg/reshare.feature [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      | permissionsRole | <reshare-permissions-role> |
+    Then the HTTP status code should be "200"
+    And for user "Carol" the space Shares should contain these entries:
+      | FolderToShare |
+    Examples:
+      | permissions-role | reshare-permissions-role |
+      | Viewer           | Viewer                   |
+      | Editor           | Viewer                   |
+      | Editor           | Editor                   |
+      | Editor           | Uploader                 |
```
Comment (SagarGi) [inline tests/acceptance/features/apiSharingNg/reshare.feature:66]: can other combination be made?

File: tests/acceptance/features/apiSharingNg/reshare.feature (hunk 1 above)
Comment (PrajwolAmatya) [reply tests/acceptance/features/apiSharingNg/reshare.feature:39]: When a user receives a share as Viewer role, he/she cannot reshare the resource to other users with the role File Editor or higher roles. In this PR only positive case is covered. Negative cases will be covered in other PR.

File: tests/acceptance/features/apiSharingNg/reshare.feature (hunk 2 above)
Comment (PrajwolAmatya) [reply tests/acceptance/features/apiSharingNg/reshare.feature:66]: same thing as in comment https://github.com/owncloud/ocis/pull/8275#discussion_r1464417031

**Review by saw-jan — APPROVED** [review]
LGTM

**Pull request merged**